internal/provider/resource_gitlab_branch_protection.go

```diff
@@ -169,30 +169,30 @@ func resourceGitlabBranchProtectionRead(ctx context.Context, d *schema.ResourceD
 	d.Set("project", project)
 	d.Set("branch", pb.Name)
 
-	pushAccessLevels := convertAllowedAccessLevelsToBranchAccessDescriptions(pb.PushAccessLevels)
-	if len(pushAccessLevels) > 0 {
-		if err := d.Set("push_access_level", pushAccessLevels[0].AccessLevel); err != nil {
+	if pushAccessLevel, err := firstValidAccessLevel(pb.PushAccessLevels); err == nil {
```

Comment:
There's a change in behavior here. Previously if no access level was found, it would not set the attribute. Now, due to the `flattenNonZeroBranchAccessDescriptions` change, it is returning a "no valid access level found" error.

I think this is ok because `push_access_level` and `merge_access_level` are required attributes. Feel free to resolve if you agree.

Reply:
Yes, that is my assumption, too.